core/src/main/java/com/redhat/lightblue/client/request/AbstractLightblueDataExecutionRequest.java

```diff
@@ -0,0 +1,37 @@
+package com.redhat.lightblue.client.request;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.redhat.lightblue.client.Execution;
+
+public abstract class AbstractLightblueDataExecutionRequest extends AbstractLightblueDataRequest {
+
+    public AbstractLightblueDataExecutionRequest(String entityName, String entityVersion) {
+        super(entityName, entityVersion);
+    }
+
+    public AbstractLightblueDataExecutionRequest(String entityName) {
+        super(entityName);
+    }
+
+    private Execution execution = null;
+
+    public AbstractLightblueDataExecutionRequest execution(Execution execution) {
+        this.execution = execution;
+        return this;
+    }
+
+    private void appendExecutionToJson(ObjectNode node) {
+        if (execution != null)
+            node.set("execution", execution.toJson());
+    }
+
+    @Override
+    public JsonNode getBodyJson() {
+        ObjectNode node = JsonNodeFactory.instance.objectNode();
```

Comment:
Why not push creation of the object node to super, and only add the execution here?